core/psh/sysexec.c

```diff
@@ -25,6 +27,63 @@ void psh_sysexecinfo(void)
 }
 
 
+static int psh_sysexec_wldargmatch(char *wld, int argc, char **argv, char wldend, int argmaxlen)
+{
+	int len = 0, ii;
+	char *currc = wld, *nextc, *cmdend;
+
+	cmdend = strchr(wld, wldend);
+	currc = wld;
+	for (ii = 0; ii < argc; ii++) {
+
+		if ((nextc = strchr(currc, ' ')) == NULL || nextc > cmdend)
+			nextc = cmdend;
+		if ((nextc == cmdend) ^ (ii == (argc - 1))) /* exits if: last argc XOR last wldcrd argument */
```

Comment:
`xor` is not really readable - maybe it would be better to break the loop as soon as one of the commands run out and check if both of them are at the end after the `for` body? (and return -1 then)

Reply:
Because we have knowledge that one command ends i wanted to break the loop before comparison operation (which is heaviest part of function with all `memcmp` and `strlen`) thus xor usage (also code is shorter)

I was fixed on checking that xor statement before main arguments equality check. Moving it after main check allows for usage of `or`. A little more code, but simpler code. Thanks